src/analysis/processing/qgsalgorithmzonalhistogram.cpp

```diff
@@ -0,0 +1,323 @@
+/***************************************************************************
+                         qgsalgorithmzonalhistogram.cpp
+                         ---------------------
+    begin                : May, 2018
+    copyright            : (C) 2018 by Mathieu Pellerin
+    email                : nirvn dot asia at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmzonalhistogram.h"
+#include "qgsgeos.h"
+#include "qgslogger.h"
+
+///@cond PRIVATE
+
+QString QgsZonalHistogramAlgorithm::name() const
+{
+  return QStringLiteral( "zonalhistogram" );
+}
+
+QString QgsZonalHistogramAlgorithm::displayName() const
+{
+  return QObject::tr( "Zonal histogram" );
+}
+
+QStringList QgsZonalHistogramAlgorithm::tags() const
+{
+  return QObject::tr( "count,area,statistics" ).split( ',' );
```

Comment:
i'd put "raster" in here too - otherwise searching for "Raster count" etc won't return results